app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,11 @@
+class AdminController < ApplicationController
+  before_action :check_for_admin
+
+  def check_for_admin
```

Comment:
All user-only actions are now in controllers that inherit from AdminController. This prevents non-authenticated people from creating/deleting articles